src/System.Private.CoreLib/shared/System/Number.BigInteger.cs

```diff
@@ -721,30 +721,10 @@ public void ShiftLeft(uint shift)
                 }
             }
 
-            [StructLayout(LayoutKind.Sequential, Pack = 1)]
-            private struct BlocksBuffer
+            private uint* GetBlocksPointer()
             {
-                private fixed uint _blocks[MaxBlockCount];
-                
-                public ref uint this[int index]
-                {
-                    get
-                    {
-                        Debug.Assert(unchecked((uint)(index)) <= MaxBlockCount);
-                        return ref Unsafe.Add(ref GetPinnableReference(), index);
-                    }
-                }
-
-                public ref uint GetPinnableReference()
-                {
-                    var pThis = Unsafe.AsPointer(ref this);
-                    return ref Unsafe.AsRef<uint>(pThis);
-                }
-
-                public uint* GetPointer()
-                {
-                    return (uint*)(Unsafe.AsPointer(ref this));
-                }
+                // This is safe to do since we are a ref struct
+                return (uint*)(Unsafe.AsPointer(ref _blocks[0]));
```

Comment:
It would be nice if C# didn't require us to have this workaround. Ideally it would recognize that _blocks (or any of the other fields) can't move because they are part of a `ref struct`.

Reply:
CC. @jaredpar

Reply:
We already have a championed issue for this here: https://github.com/dotnet/csharplang/issues/1792